pkg/vgmanager/vgmanager_controller.go

```diff
@@ -381,3 +381,21 @@ func (r *VGReconciler) getNewNodeStatus(status *lvmv1alpha1.VGStatus) *lvmv1alph
 	setStatus(status, vgNodeStatus)
 	return vgNodeStatus
 }
+
+func (r *VGReconciler) matchesMyNode(ctx context.Context, selector *corev1.NodeSelector) (bool, error) {
+
+	node := &corev1.Node{}
+	err := r.Client.Get(ctx, types.NamespacedName{Name: r.NodeName}, node)
+	if err != nil {
+		return false, err
+	}
+	nodeMatches, err := NodeSelectorMatchesNodeLabels(node, selector)
+	if err != nil {
+		return false, err
+	}
+
+	if !nodeMatches {
```

Comment:
This check might not be needed. Function can return `nodeMatches` variable.

Reply:
done.